ring/batch.go

```diff
@@ -4,13 +4,44 @@ package ring
 
 import (
 	"context"
+	"errors"
 	"fmt"
 	"sync"
 
 	"go.uber.org/atomic"
 	"google.golang.org/grpc/status"
 )
 
+type Error interface {
```

Comment:
I'd rename this to something like ClientServerError. Also why do we need the embedded `error` interface at all?